cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -381,14 +434,19 @@ private boolean isReplicaMovementDone(Cluster cluster, TopicPartition tp, Execut
         case IN_PROGRESS:
           return destinationExists && !sourceExists;
         case ABORTING:
-          return !destinationExists && sourceExists;
+          // There could be a race condition that when we abort a task, it is already completed.
+          // in that case, we treat it as aborted as well.
+          return (!destinationExists && sourceExists) || (destinationExists && !sourceExists);
         case DEAD:
-          return !destinationExists && !sourceExists;
+          return true;
         default:
           throw new IllegalStateException("Should never be here. State " + task.state());
       }
     }
 
+    /**
+     * A replica deletion is done when the the source node is no longer in the replica list.
```

Comment:
Could there be a case, where the source exists, but the task state is `DEAD`?

Reply:
I don't think so.